rust/stackable-cockpit/src/platform/service.rs

```diff
@@ -45,40 +45,45 @@ pub async fn get_endpoints(
     object_name: &str,
     object_namespace: &str,
 ) -> Result<IndexMap<String, String>, Error> {
+    let mut endpoints = IndexMap::new();
+
     let list_params =
         ListParams::from_product(product_name, Some(object_name), k8s::ProductLabel::Name);
 
     let listeners = kube_client
         .list_listeners(Some(object_namespace), &list_params)
-        .await
-        .context(KubeClientFetchSnafu)?;
-
-    let mut endpoints = IndexMap::new();
-    for listener in &listeners {
-        let Some(display_name) = display_name_for_listener_name(&listener.name_any(), object_name)
-        else {
-            continue;
-        };
-        let Some(listener_status) = &listener.status else {
-            continue;
-        };
-
-        for address in listener_status.ingress_addresses.iter().flatten() {
-            for port in &address.ports {
-                let text = format!("{display_name}-{port_name}", port_name = port.0);
-                let endpoint_url = endpoint_url(&address.address, *port.1, port.0);
-                endpoints.insert(text, endpoint_url);
+        .await;
+
+    // In case the listener-operator is not installed, this will return a 404. We should not fail, as this causes
+    // stackablectl to fail with ApiError 404 on clusters without listener-operator.
+    if let Ok(listeners) = listeners {
```

Comment:
We currently already get a warning (which I was not able to get rid of)
```
➜  stackable-cockpit git:(fix/no-listener-installed) ✗ cargo r -p stackablectl -- stacklet list
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
     Running `target/debug/stackablectl stacklet list`
   WARN  Unsuccessful data error parse: 404 page not found

    at /home/sbernauer/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kube-client-0.87.1/src/client/mod.rs:445


┌───────────┬───────────┬───────────┬──────────────────────┬─────────────────────────────────┐
│ PRODUCT   ┆ NAME      ┆ NAMESPACE ┆ ENDPOINTS            ┆ CONDITIONS                      │
╞═══════════╪═══════════╪═══════════╪══════════════════════╪═════════════════════════════════╡
│ zookeeper ┆ simple-zk ┆ default   ┆ zk  172.18.0.2:30840 ┆ Available, Reconciling, Running │
└───────────┴───────────┴───────────┴──────────────────────┴─────────────────────────────────┘

Use "stackablectl stacklet credentials [OPTIONS] <PRODUCT_NAME> <STACKLET_NAME>" to display credentials for deployed stacklets.
```

Honestly I don't like it, because a cluster without listener-op is still valid and IMHO not worth a warning. But I don't care, we just need to be able to implement it :D

Reply:
Alright, good enough for now I guess.

We could adjust the tracing filter to get rid of the warning, but to be honest, that doesn't sound like a great idea. I guess we have to live with that warning until the listener is fully rolled out.